docs/src/modules/components/ApiPage/processors/classes.ts

```diff
@@ -0,0 +1,60 @@
+import { PropsTranslations, ComponentApiContent } from '@mui-internal/api-docs-builder';
+import { Translate } from '@mui/docs/i18n';
+import kebabCase from 'lodash/kebabCase';
+import type { TableOfContentsParams } from 'docs/src/modules/components/ApiPage';
+
+export interface ClassDefinition {
+  className: string;
+  key: string;
+  hash: string;
+  description?: string;
+  isGlobal?: boolean;
+  isDeprecated?: boolean;
+  deprecationInfo?: string;
+}
+
+export type GetCssToCParams = {
+  classes: ClassDefinition[];
+  t: Translate;
+  hash?: string;
+};
+
+export const getClassesToC = ({ classes, t, hash }: GetCssToCParams): TableOfContentsParams[] =>
+  !classes || classes.length === 0
+    ? []
+    : [
+        {
+          text: t('api-docs.classes'),
+          hash: hash ?? 'classes',
+          children: [
+            ...classes.map(({ key, hash: classeHash }) => ({
+              text: key,
+              hash: classeHash,
+              children: [],
+            })),
+          ],
+        },
+      ];
+
+export interface ClassesApiProcessorParams {
+  componentClasses: ComponentApiContent['classes'];
+  classDescriptions: PropsTranslations['classDescriptions'];
+  componentName: string;
+}
+
+export function classesApiProcessor(params: ClassesApiProcessorParams): ClassDefinition[] {
+  const { componentClasses, classDescriptions, componentName } = params;
+
+  return componentClasses.map((classDefinition) => {
+    return {
+      ...classDefinition,
+      description:
+        classDescriptions[classDefinition.key]?.description
+          ?.replace(/{{conditions}}/, classDescriptions[classDefinition.key].conditions!)
```

Comment:
Is there any way to get rid of this non-null assertion operator? It removes type safety and essentially and upstream bug can turn this replacement into `'undefined'` without a warning.
If it's hard to guarantee type-wise and `undefined` here would signify a programming error. An `invariant()` would turn this into a build time error instead of quietly starting to replace strings with `'undefined'`.

Reply:
I went with the following solution
```ts
    if (!conditions && description.search(/{{conditions}}/) !== -1) {
      throw Error(
        `In ${componentName} the class "${classDefinition.className}" description with "{{conditions}}" but without \`conditions\` to replace it.`,
      );
    }
```

It adds the runtime error, but does not fix the typing, because the fact that  `conditions` is defined depends on the presence of `{{conditions}}` in the description, which might be hard to define with types

Reply:
The way I would write this is to replace only within the check of whether the description contains `{{conditions}}`

```tsx
if (!conditions && description.includes(/{{conditions}}/)) {
  throw new Error('...')
}

// This is suboptimal, a type checker can't relate the previous condition with the following replacement

return {
  description: description.replace(/{{conditions}}/, conditions!),
}
```
The solution requires a refactor
```tsx
// make description a mutable variable

if (description.includes(/{{conditions}}/)) {
  // Use an invariant to assert the condition, this would be a programming error
  invariant(conditions, 'conditions should be defined if description includes {{conditions}}')
  descripton = description.replace(/{{conditions}}/, conditions)
}

return {
  description
}
```

But again, this is all polish. Doesn't have to be in scope for this PR

Reply:
Thanks for the detailed explanation 👍